src/js/actions/layers.js

```diff
@@ -766,18 +834,31 @@ define(function (require, exports) {
                 .unshift(documentRef)
                 .toArray();
 
-        var expandPlayObject = layerLib.setGroupExpansion(layerRefs, !!expand),
-            expansionPromise = descriptor.playObject(expandPlayObject),
+        var expandPlayObject = layerLib.setGroupExpansion(layerRefs, !!expand);
+
+        playObjects.push(expandPlayObject);
+
+        var expansionPromise = descriptor.batchPlayObjects(playObjects),
             dispatchPromise = this.dispatchAsync(events.document.SET_GROUP_EXPANSION, {
                 documentID: document.id,
                 layerIDs: collection.pluck(layers, "id"),
+                selected: collection.pluck(layersToSelect, "id"),
+                deselected: collection.pluck(layersToDeselect, "id"),
                 expanded: expand
             });
 
-        return Promise.join(expansionPromise, dispatchPromise);
+        return Promise.join(expansionPromise, dispatchPromise, function () {
+            if (layersToSelect.length > 0) {
```

Comment:
We could have selected new layers which are uninitialized, so we transfer at the end of the action to `initializeLayers`.